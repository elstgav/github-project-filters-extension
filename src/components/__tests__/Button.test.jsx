import Button from 'components/Button'
import EnzymeTestWrapper from 'test/support/EnzymeTestWrapper'

const test = new EnzymeTestWrapper(Button)

describe('Button', () => {
  beforeEach(() => {
    test.props = { children: 'Foo' }
  })

  it('displays a <button> element', () => {
    expect(test.Button.find('button')).toHaveLength(1)
    expect(test.Button.props().className).toBe('btn btn-default')
    expect(test.Button.text()).toBe('Foo')
    expect(test.Button).toMatchSnapshot()
  })

  it('has a configurable className', () => {
    test.props.className = 'foo'
    expect(test.Button.props().className).toBe('btn btn-default foo')
  })

  describe('when isActive', () => {
    beforeEach(() => {
      test.props.isActive = true
    })

    it('adds a selected class', () => {
      expect(test.Button.props().className).toBe('btn btn-default selected')
    })
  })
})
