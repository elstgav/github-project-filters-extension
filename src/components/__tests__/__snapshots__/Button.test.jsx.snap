// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays a <button> element 1`] = `
<button
  className="btn btn-default"
  type="button"
>
  Foo
</button>
`;
